### PR TITLE
Couple fixes for creality boards

### PR DESCRIPTION
**Pull request by EvilGremlin**
Correct LCD_SERIAL_PORT and nobootloader env for RC chip (because nothing fun fits with insane 64k boot)

### CODE REVIEW

File: Marlin/src/inc/Conditionals_adv.h [hunk 1]
```diff
@@ -1240,7 +1240,7 @@
 #if LCD_IS_SERIAL_HOST && !defined(LCD_SERIAL_PORT)
   #if MB(BTT_SKR_MINI_E3_V1_0, BTT_SKR_MINI_E3_V1_2, BTT_SKR_MINI_E3_V2_0, BTT_SKR_MINI_E3_V3_0, BTT_SKR_E3_TURBO, BTT_OCTOPUS_V1_1)
     #define LCD_SERIAL_PORT 1
-  #elif MB(CREALITY_V24S1_301, CREALITY_V24S1_301F4, CREALITY_V423, MKS_ROBIN, PANOWIN_CUTLASS, KODAMA_BARDO)
+  #elif MB(CREALITY_V24S1_301, CREALITY_V24S1_301F4, BOARD_CREALITY_F401RE, CREALITY_V423, MKS_ROBIN, PANOWIN_CUTLASS, KODAMA_BARDO)
```
Comment (thisiskeithb) [inline Marlin/src/inc/Conditionals_adv.h]: You’ll need to drop the `BOARD_` portion of the define since that is already handled with the `MB()` macro.

Comment (EvilGremlin) [reply Marlin/src/inc/Conditionals_adv.h]: ouch i'm blind thanks

**Review by classicrocker883 — CHANGES_REQUESTED** [review]
what if we took `board_build.rename   = firmware-{date}-{time}.bin`, added it to **platformio.ini** `[env]`?

that way it will encompass all firmware to be "firmware-$date-$time.bin", but if the default `[env:  ]` already has `board_build.rename = ` in its **.ini** then it will resort to that.

**Comment by ellensp** [comment]
@classicrocker883 creality is not the be-all and end-all of all things. Their defaults should not be the standard default

There are 703 supported boards, of which there are only 42 Creality board. Not all of which need this format.

**Comment by ThomasToka** [comment]
> what if we took `board_build.rename = firmware-{date}-{time}.bin`, added it to **platformio.ini** `[env]`?
> 
> that way it will encompass all firmware to be "firmware-$date-$time.bin", but if the default `[env: ]` already has `board_build.rename = ` in its **.ini** then it will resort to that.

```
BIN_NAME = Path(target[0].path).replace(Path(target[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S'))))
ELF_NAME = Path(source[0].path).replace(Path(source[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S').replace('.bin', '.elf'))))
print("FIRMWARE ELF: %s" % ELF_NAME)
print("FIRMWARE BIN: %s" % BIN_NAME)
```

i do something like this in my fork to unify the names of bin and elf.

**Comment by classicrocker883** [comment]
> @classicrocker883 creality is not the be-all and end-all of all things. Their defaults should not be the standard default
> 
> There are 703 supported boards, of which there are only 42 Creality board. Not all of which need this format.

that may be true, however, as I stated 
>but if the default [env:  ] already has board_build.rename =  in its .ini then it will resort to that.

meaning, it will overwrite it. I tested it and it works. I just think having the default *firmware.bin* name should at least include the date/time. and any other board which doesn't want that will go unchanged being renamed something else.  

I also just realized that `buildroot/share/PlatformIO/scripts/offset_and_rename.py` should be placed in there as well.

but even similarly how about placing:

`pre:buildroot/share/PlatformIO/scripts/random-bin.py` with the other `extra_scripts` under `[common]`? I mean this is only being used by literally two - the BTT SKR Mini E3 V3.0 and V3.0.1.  
maybe even completely omit random-bin.py and replace it with `board_build.rename`?

**Comment by classicrocker883** [comment]
> > what if we took `board_build.rename = firmware-{date}-{time}.bin`, added it to **platformio.ini** `[env]`?
> > that way it will encompass all firmware to be "firmware-$date-$time.bin", but if the default `[env: ]` already has `board_build.rename = ` in its **.ini** then it will resort to that.
> 
> ```
> BIN_NAME = Path(target[0].path).replace(Path(target[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S'))))
> ELF_NAME = Path(source[0].path).replace(Path(source[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S').replace('.bin', '.elf'))))
> print("FIRMWARE ELF: %s" % ELF_NAME)
> print("FIRMWARE BIN: %s" % BIN_NAME)
> ```
> 
> i do something like this in my fork to unify the names of bin and elf.

since this file was updated, I mean its fine this way, but I changed it to be more compatible with the more recent version.

```py
            _newtarget = Path(target[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S')))
            _newsource = Path(source[0].dir.path, datetime.now().strftime(new_name.replace('{date}', '%Y%m%d').replace('{time}', '%H%M%S').replace('.bin', '.elf')))
            BIN_NAME = Path(target[0].path).replace(_newtarget)
            ELF_NAME = Path(source[0].path).replace(_newsource)
            env['PROGNAME'] = path.splitext(_newtarget)[0]
            print("FIRMWARE ELF: %s" % ELF_NAME)
            print("FIRMWARE BIN: %s" % BIN_NAME)
```

**Comment by thisiskeithb** [comment]
@classicrocker883 Creality boards make up less than 6% total supported boards and an even smaller subset needs a custom/unique firmware name.

It makes no sense to change the default behavior to support that. Feel free to update your fork since you seem to be primarily focused on those types of boards.

**Comment by classicrocker883** [comment]
> @classicrocker883 Creality boards make up less than 6% total supported boards and an even smaller subset needs a custom/unique firmware name.
> 
> It makes no sense to change the default behavior to support that. Feel free to update your fork since you seem to be primarily focused on those types of boards.

that is not at all what I am saying.

did I not make my self clear?

I'm not proposing changing default behavior to support one board or another.

all I am saying is that we add **date/time** to the end, which so happens to be like for Creality boards - yes, but not because of that reason.

at the same time, any board which has its own "rename" won't be affected in the end...

perhaps I can create a PR to illustrate my point.

**Comment by ellensp** [comment]
and any that expect exactly "firmware.bin" will break! potentially some 650 boards. which ones?  no one knows, each and every environment would need to be tested on real hardware

**Comment by thisiskeithb** [comment]
> perhaps I can create a PR to illustrate my point.

Please don’t.

`firmware.bin` **is** the default firmware name and should continue to remain the standard. Boards that need the firmware binary renamed will work from that.

There is **no** reason to make the Creality / date format standard.

**Comment by ThomasToka** [comment]
> > perhaps I can create a PR to illustrate my point.
> 
> Please don’t.
> 
> `firmware.bin` **is** the default firmware name and should continue to remain the standard. Boards that need the firmware binary renamed will work from that.
> 
> There is **no** reason to make the Creality / date format standard.

There is more than one reason for this.

- versioning
- documentation
- oversight

As always users decide what is needed and what not. So assuming there is no reason is wrong.

Its requested so its needed. Simple as that.

Maybe a more finer Integration is needed. 

I also did this cause marlins default behavior is simply too bad with this.

**Comment by EvilGremlin** [comment]
I only see users of one brand asking for that, no one else. Which makes no sense as they already have this renaming. 
And WTF with my small PR becoming a place for completely irrelevant discussion?

**Comment by thinkyhead** [comment]
> WTF with my small PR becoming a place for completely irrelevant discussion?

I know, right??

I'll be happy to add an option to Auto Build Marlin to apply a datestamp to the built firmware, but I'm not sure it's a good idea to make it the standard, because AFAIK the vast majority of boards will only look for `firmware.bin` on the SD card, and will ignore a BIN file with another name. Of course, legacy documentation all over the internet often refers specifically to `firmware.bin` so that is what users will be looking for. It's not clear what happens if you rename the `firmware.hex` file as part of the build, but since that file doesn't need to be read from SD it should be ok to call it anything.

Another alternative would be to add an "archive" checkbox to Auto Build Marlin. In this case the firmware build (including ELF, etc.) would be copied into a sub-folder with a datestamp. That would accomplish the archiving while also preserving the expected naming of the `firmware.bin` file for the given board/bootloader target.

**Pull request merged**